packages/cli/src/commands/spaces/topology.ts

```diff
@@ -0,0 +1,114 @@
+import {Command, flags} from '@heroku-cli/command'
+import {Args, ux} from '@oclif/core'
+import * as Heroku from '@heroku-cli/schema'
+import heredoc from 'tsheredoc'
+import color from '@heroku-cli/color'
+
+export type SpaceTopology = {
+  version: number,
+  apps: Array<{
+    id?: string
+    domains: string[]
+    formations: Array<{
+      process_type: string
+      dynos: Array<{
+        number: number
+        private_ip: string
+        hostname: string
+      }>
+    }>
+  }>
+}
+
+export default class Topology extends Command {
+  static topic = 'spaces';
+  static description = 'show space topology';
+  static flags = {
+    space: flags.string({char: 's', description: 'space to get topology of'}),
+    json: flags.boolean({description: 'output in json format'}),
+  };
+
+  static args = {
+    space: Args.string({hidden: true}),
+  };
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(Topology)
+    const spaceName = flags.space || args.space
+    if (!spaceName) {
+      ux.error(heredoc(`
+        Error: Missing 1 required arg:
+        space
+        See more help with --help
+      `))
+    }
+
+    const {body: topology} = await this.heroku.get<SpaceTopology>(`/spaces/${spaceName}/topology`)
+    let appInfo: Heroku.App[] = []
+    if (topology.apps) {
+      appInfo = await Promise.all(topology.apps.map(async topologyApp => {
+        const {body: app} = await this.heroku.get<Heroku.App>(`/apps/${topologyApp.id}`)
+        return app
+      }))
+    }
+
+    this.render(topology, appInfo, flags.json)
+  }
+
+  protected render(topology: SpaceTopology, appInfo: Heroku.App[], json: boolean) {
+    if (json) {
+      ux.styledJSON(topology)
+    } else if (topology.apps) {
+      topology.apps.forEach(app => {
+        let formations: string[] = []
+        let dynos: string[] = []
+        if (app.formations) {
+          app.formations.forEach(formation => {
+            formations.push(formation.process_type)
+            if (formation.dynos) {
+              formation.dynos.forEach(dyno => {
+                const dynoS = [`${formation.process_type}.${dyno.number}`, dyno.private_ip, dyno.hostname].filter(Boolean)
+                dynos.push(dynoS.join(' - '))
+              })
+            }
+          })
+        }
+
+        const domains = app.domains.sort()
+        formations = formations.sort()
+        dynos = dynos.sort((a, b) => {
```

Comment:
These assignments after sorting are superfluous since sorting is done in place. When I see these in the legacy codebase, I usually remove them so be more idomatic.

Reply:
Oh yeah, nice catch. We should probably have an eslint rule around this.